tests/engine/test_flow_runner.py

```diff
@@ -1548,3 +1548,33 @@ def add(x):
 
     ## to be safe, ensure '5' isn't in the logs
     assert len([log.message for log in caplog.records if "99" in log.message]) == 0
+
+
+def test_flow_runner_provides_consistent_extra_context():
```

Comment:
we might want to add a docstring to this test that it's really about the dask executor, because a dev who edits a key from `extra_context` might edit this test to pass without realizing it affects the task keys in dask.  Sorry to be nitpicky, I just think we have a lot of tests that are testing _around_ the intended behavior without testing it directly 
![image](https://user-images.githubusercontent.com/13255838/85185533-2ae18180-b249-11ea-8cbf-67f75da4ee41.png)

Reply:
Since there is a test for the dask executor that will fail if those change, changing one without fixing the other shouldn't be possible. But sure. I feel like that image describes much of the current tests for prefect.

Reply:
Interesting, I find that view contradictory - I think that this key thing has a lot of scattered hyper-specific unit tests that are still easy to make pass without ever knowing whether the real objective is achieved via a test (which we used to have).

Reply:
switched to an integration test